realm/realm-library/src/objectServer/java/io/realm/internal/objectstore/OsMongoCollection.java

```diff
@@ -90,99 +111,142 @@ protected Long mapSuccess(Object result) {
             }
         };
 
-        // no filter means count all
-        String filterString = (filter == null) ?
-                JniBsonProtocol.encode(new Document(), codecRegistry) :
-                JniBsonProtocol.encode(filter, codecRegistry);
+        String filterString = JniBsonProtocol.encode(filter, codecRegistry);
         int limit = (options == null) ? 0 : options.getLimit();
 
         nativeCount(nativePtr, filterString, limit, callback);
 
         return ResultHandler.handleResult(success, error);
     }
 
-    public Collection<DocumentT> find() {
-        return find(new Document());
-    }
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
I will remove these separators. I added them for temporary convenience to quickly see the different operation blocks.